iocore/net/SSLNetVConnection.cc

```diff
@@ -1329,25 +1322,24 @@ SSLNetVConnection::sslClientHandShakeEvent(int &err)
     return EVENT_ERROR;
 
   case SSL_ERROR_SYSCALL:
-    err = errno;
     SSL_INCREMENT_DYN_STAT(ssl_error_syscall);
     Debug("ssl.error", "SSLNetVConnection::sslClientHandShakeEvent, syscall");
-    TraceIn(trace, get_remote_addr(), get_remote_port(), "SSL client handshake Syscall Error: %s", strerror(errno));
+    TraceIn(trace, get_remote_addr(), get_remote_port(), "SSL client handshake Syscall Error: %s", strerror(err));
     return EVENT_ERROR;
     break;
 
   case SSL_ERROR_SSL:
   default: {
-    err = (errno) ? errno : -ENET_CONNECT_FAILED;
+    err = (err) ? err : -ENET_CONNECT_FAILED;
```

Comment:
`err` should be introduced to save `errno` but is changed here. It makes someone create a PR like "save err ASAP"

Reply:
IMO, `err = (errno) ? errno : -ENET_CONNECT_FAILED;` is here because someone found the `errno` is overwrote to 0.

Reply:
Does that mean we can remove the line (because `errno` should be saved in `err`)?
If `err` is also overwritten, we're still doomed to refer a wrong error code.

Reply:
@maskit  IMO, It is better to set ```err``` to ```-ENET_CONNECT_FAILED``` because ```SSL_ERROR_SSL``` does not means any syscall failed.

Reply:
What about default case?